test/controllers/posts_controller_test.rb

```diff
@@ -3,6 +3,7 @@
 class PostsControllerTest < ActionDispatch::IntegrationTest
   setup do
     @post = posts(:one)
+    @user = users(:one)
```

Comment:
what is the (:one) doing here? Just for my own learning.

Reply:
If you go into fixtures/user.yml and fixtures/posts.yml you will find some database entries specified.

When we run tests with Rails default UnitTest... it uses a test database that has nothing in it.

So before I run the tests I need a post and a user. The code above makes the test database load user :one and post :one before every single test. If there are 7 tests it will load them 7 times.

So the :one just specifies which post and which user I want.

Reply:
Sounds good! Thanks